runner/client/src/mill/runner/client/JansiLoader.java

```diff
@@ -0,0 +1,50 @@
+package mill.runner.client;
+
+import java.io.File;
+import java.io.IOException;
+import org.fusesource.jansi.internal.OSInfo;
+
+final class JansiLoader {
+
+  private String jansiVersion;
+
+  JansiLoader(String jansiVersion) {
+    this.jansiVersion = jansiVersion;
+  }
+
+  private String jansiLibPathInArchive() {
+    return "org/fusesource/jansi/internal/native/" + OSInfo.getNativeLibFolderPathForCurrentOS()
+        + "/" + System.mapLibraryName("jansi").replace(".dylib", ".jnilib");
+  }
+
+  File tryLoadFast() {
+    File archiveCacheLocation;
+    try {
+      archiveCacheLocation = coursier.paths.CachePath.defaultArchiveCacheDirectory();
+    } catch (IOException ex) {
+      throw new RuntimeException(ex);
+    }
+    File jansiLib = new File(
+        archiveCacheLocation,
+        "https/repo1.maven.org/maven2/org/fusesource/jansi/jansi/" + jansiVersion + "/jansi-"
```

Comment:
These hardcoded references to `repo1.maven.org` are potentially problematic, e.g. some folks work in companies with access to maven central blocked to force their developers to use their internal mirror. Is there some way we can make this obey the `COURSIER_REPOSITORIES` configuration to support such use cases?

Reply:
e.g. see https://github.com/com-lihaoyi/mill/issues/3408

Reply:
That's a good point… In a first time, maybe we could detect if the default repositories of coursier were overridden or not. If they're overridden, keep doing what we were doing before (calling `tput` every n-th mlliseconds). If they're not, then proceed to do what the PR currently does. Maybe we could also check if `~/.ivy2/local` contains the version of jansi that we use, and do as if repositories were overridden. That way, we could merge now and benefit from this PR in many cases already.

In a second time, in a subsequent PR, I could manage to pull some more coursier logic in Java classes (in the coursier codebase), so that we could stay in Java land here, even when the coursier repositories are overridden.

Reply:
Could we bundle the relevant binaries with Mill and unpack them from the Mill executable by reading them from the classpath? That would save us the whole trip-to-maven-central thing and make it nicely self-contained

Reply:
Writing it out to `~/.cache/mill` sounds reasonable